src/store/RelayRecordStore.js

```diff
@@ -470,17 +470,22 @@ class RelayRecordStore {
       storageKey,
       dataID
     );
-    return field.map((item, ii) => {
+    const linkedIDs = [];
+    field.forEach((item, ii) => {
+      const id = item.__dataID__;
```

Comment:
it's possible that the value could be an array of scalars, i'm surprised Flow accepted this. Can you revert this part to only access `__dataID__` after confirming `item` is a non-null object?